src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -106,13 +117,44 @@ private static string GetDotNetBasePath(string workingDirectory)
 
             var outputString = string.Join(Environment.NewLine, lines);
 
+            var lineSdkIndex = lines.FindIndex(line => line.Contains(".NET Core SDKs installed"));
+
+            if (lineSdkIndex != -1)
+            {
+                lineSdkIndex++; 
+
+                while (lineSdkIndex < lines.Count && !lines[lineSdkIndex].Contains(".NET Core runtimes installed"))
+                {
+                    var ma = SdkRegex.Match(lines[lineSdkIndex]);
+
+                    if (!ma.Success)
+                        break;
+
+                    var version = ma.Groups[1].Value.Trim();                     
+                    var path = ma.Groups[2].Value.Trim();
+
+                    path = Path.Combine(path, version) + "\\";
+
+                    //basePaths.Add(path);
+
+                    // We insert at index 0 so that instance list will be sorted descending so that instances.FirstOrDefault() 
+                    // will always return the latest installed version of dotnet SDK 
+                    basePaths.Insert(0, path);      
+
+                    lineSdkIndex++;
+                }
+            }
+
             var matched = DotNetBasePathRegex.Match(outputString);
             if (!matched.Success)
```

Comment:
Move this check before parsing sdks, since it would keep us from doing extra work.

Reply:
Fixed.